core/repository/consensus/merkle_transaction_repository.hpp

```diff
@@ -26,58 +26,58 @@ limitations under the License.
 
 namespace merkle_transaction_repository {
 
-    // Think in progress ^^;
-//struct MerkleNode {
-//    std::string hash;
-//    std::string parent;
-//    std::string leftChild;
-//    std::string rightChild;
-//
-//    MerkleNode() {
-//
-//    }
-//
-//    MerkleNode(
-//        const std::string hash,
-//        const std::string parent,
-//        const std::string leftChild,
-//        const std::string rightChild
-//    ):
-//            hash(hash),
-//            parent(parent),
-//            left(leftChild),
-//            right(rightChild)
-//    {}
-//
-//    bool isRoot() {
-//        return parent.empty();
-//    }
-//
-//    bool isLeaf() {
-//        return leftChild.empty() && rightChild.empty();
-//    }
-//
-//    using Object = json_parse::Object;
-//    using Rule = json_parse::Rule;
-//    using Type = json_parse::Type;
-//    Object dump() {
-//        Object obj = Object(Type::DICT);
-//        obj.dictSub["hash"] =  Object(Type::STR, hash);
-//        obj.dictSub["parent"] =  Object(Type::STR, parent);
-//        obj.dictSub["leftChild"] =  Object(Type::STR, leftChild);
-//        obj.dictSub["rightChild"] =  Object(Type::STR, rightChild);
-//        return obj;
-//    }
-//
-//    static Rule getJsonParseRule() {
-//        Rule obj = Rule(Type::DICT);
-//        obj.dictSub["hash"] =  Rule(Type::STR);
-//        obj.dictSub["parent"] =  Rule(Type::STR);
-//        obj.dictSub["leftChild"] = Rule(Type::STR);
-//        obj.dictSub["rightChild"] = Rule(Type::STR);
-//        return obj;
-//    }
-//};
+
+struct MerkleNode {
+    std::string hash;
+    std::string parent;
+    std::string left;
+    std::string right;
+
+    MerkleNode() {
+
+    }
+
+    MerkleNode(
+        const std::string hash,
+        const std::string parent,
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Function parameter 'parent' should be passed by reference.](https://www.codacy.com/app/soramitsu/iroha/file/4371437967/issues/source?bid=3712491&fileBranchId=3879509#l42)